test/threads_exec.jl

```diff
@@ -845,3 +845,32 @@ fib34666(x) =
         f(x)
     end
 @test fib34666(25) == 75025
+
+function jitter_channel(f, k, delay, ntasks, schedule)
+    x = Channel(ch -> foreach(i -> put!(ch, i), 1:k), 1)
+    y = Channel(k) do ch
+        g = i -> begin
+            iseven(i) && sleep(delay)
+            put!(ch, f(i))
+        end
+        Threads.foreach(g, x, schedule; ntasks=ntasks)
+    end
+    return y
+end
+
+@testset "Threads.foreach(f, ::Channel)" begin
```

Comment:
I think it'd be nice to test the default keyword argument (e.g., a typo may can be introduced with some edits in the future). Maybe something like

```julia
ys = Channel() do ys
    Threads.foreach(Channel(xs -> foreach(i -> put!(xs, i), 1:3))) do x
        put!(ys, x)
    end
end
@test sort!(collect(ys)) == 1:3
```

Reply:
Done.